### PR TITLE
Status quo

### PR DESCRIPTION
**Pull request by sd-lueckel**
Added some basic features of stuff which should not break during development

### CODE REVIEW

**Review by mattagohni — DISMISSED** [review]

File: tests/Behat/Context/Ui/Admin/MenuContext.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+use Webmozart\Assert\Assert;
+
+class MenuContext implements Context
+{
+    /** @var MenuPageInterface */
+    private $menuPage;
+
+    /**
+     * @param MenuPageInterface $menuPage
+     *
```
Comment (michz) [inline tests/Behat/Context/Ui/Admin/MenuContext.php]: Newline too much

File: tests/Behat/Page/Admin/MenuPage.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+namespace Tests\solutionDrive\SyliusAdminSecurityThroughObscurityPlugin\Behat\Page\Admin;
+
+use Behat\Mink\Element\NodeElement;
+use Behat\Mink\Exception\ElementNotFoundException;
+use Sylius\Behat\Page\SymfonyPage;
+
+class MenuPage extends SymfonyPage implements MenuPageInterface
+{
+    /**
+     * {@inheritdoc}
```
Comment (michz) [inline tests/Behat/Page/Admin/MenuPage.php]: Really? But why not for other const and getters? Something is not consistent here

**Comment by sd-lueckel** [comment]
Review required!

File: tests/Behat/Page/Admin/MenuPage.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        } catch (ElementNotFoundException $exception) {
+            $menuEntry = null;
+        }
+        return $menuEntry;
+    }
+
+    protected function getDefinedElements()
+    {
+        $catalogMenuElements = [
+            'Catalog'                       => self::MAIN_ENTRY . ':contains("Catalog")',
```
Comment (twojtylak) [inline tests/Behat/Page/Admin/MenuPage.php]: here you can use less blank spaces

**Review by twojtylak — DISMISSED** [review]

File: tests/Behat/Context/Ui/Admin/MenuContext.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    /**
+     * @Then I should see :menuValue menu entry
+     */
+    public function iShouldSeeMenuEntry($menuValue)
+    {
+        $menuEntry = $this->menuPage->getMenuEntryByValue($menuValue);
+        Assert::notNull($menuEntry, "Could not find the menu entry '${menuValue}'");
+    }
+}
+
```
Comment (mattagohni) [inline tests/Behat/Context/Ui/Admin/MenuContext.php]: too much empty lines.

File: tests/Behat/Page/Admin/MenuPage.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            parent::getDefinedElements(),
+            $catalogMenuElements,
+            $salesMenuElements,
+            $customerMenuElements,
+            $marketingMenuElements,
+            $configurationMenuElements
+        );
+    }
+}
+
```
Comment (mattagohni) [inline tests/Behat/Page/Admin/MenuPage.php]: too much empty lines.

File: tests/Behat/Page/Admin/MenuPageInterface.php [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+namespace Tests\solutionDrive\SyliusAdminSecurityThroughObscurityPlugin\Behat\Page\Admin;
+
+use Behat\Mink\Element\NodeElement;
+use Sylius\Behat\Page\SymfonyPageInterface;
+
+interface MenuPageInterface  extends SymfonyPageInterface
+{
+    public function getMenuEntryByValue(string $menuValue): ?NodeElement;
+}
+
```
Comment (mattagohni) [inline tests/Behat/Page/Admin/MenuPageInterface.php]: too much empty lines.

**Comment by sd-lueckel** [comment]
Ready for review again!

**Review by michz — APPROVED** [review]

**Review by mattagohni — APPROVED** [review]

**Pull request merged**